### DISCUSSION TITLE
Run on website

### DISCUSSION THREAD

**Discussion by Stuntlover-TM** [rwf2/Rocket#2294] [Questions]
Hi, sorry if this seems very stupid but my brain just can't think of it right now; How would I actually run what Rocket does on my website (aka not localhost), is there some wasm file it creates in some hidden place? Thanks in advance.
(Or is this not what Rocket is made for at all?)

**Comment by robertwayne** [comment]
I'm not particularly clear on what you're asking.

If you mean how do you get Rocket to serve web pages to the outside world, like when you visit `example.com` - you just have to run the binary created with `cargo build --release` on your server *(this is in your target/release/ directory)*. The DNS for your domain should have an A record pointing to your server IP address, and you would want ports 22/443 open for HTTP/HTTPS traffic.

It is generally recommended to run the (Rocket) app behind a reverse-proxy *(for various reasons, like TLS termination, load balancing, static file caching, compression etc.)*. You would set your reverse proxy to point all traffic from `example.com` to `127.0.0.1:<your_rocket_app_port>`. I use Nginx, but there are plenty of other options like HAproxy and Apache.

Then Rocket would serve up your HTML/CSS/JS or JSON API or whatever.

**Reply by Stuntlover-TM** [reply]
Ahh yes that makes alot of sense now, thanks!

**Reply by robertwayne** [reply]
No problem.

If you have any specific questions regarding setting that up, let me know.

**Reply by Stuntlover-TM** [reply]
Oh another question, will normal exe's run on linux? (no idea what distro my server is), or do I have to specifically compile it for linux in rust when building?

**Reply by robertwayne** [reply]
Generally no. You can run an .exe file with Wine, but you should compile specifically for the OS.

There are cross-compilation solutions, but I haven't tried any of them and I believe they can be quirky. The easiest option is to clone your repo onto your server and build it fresh from there.

EDIT: Additionally, since you're on Windows, you can compile in [Ubuntu via WSL](https://docs.microsoft.com/en-us/windows/wsl/about) and then use that binary. It's more than likely your server will be running a Debian-based distro - probably Ubuntu, as I am assuming it is a VPS and (most) of them default to Ubuntu LTS.
